lock-facebook/src/main/java/com/auth0/android/facebook/FacebookSignInDelegateActivity.java

```diff
@@ -0,0 +1,39 @@
+package com.auth0.android.facebook;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.support.v7.app.AppCompatActivity;
+
+import com.facebook.FacebookSdk;
+import com.facebook.login.LoginManager;
+
+import java.util.Collections;
+import java.util.List;
+
+public class FacebookSignInDelegateActivity extends AppCompatActivity {
+
+    public static final String FACEBOOK_PERMISSIONS_EXTRA = "com.auth0.android.facebook.FACEBOOK_PERMISSIONS";
+    public static final String PROVIDER_REQUEST_CODE_EXTRA = "com.auth0.android.facebook.PROVIDER_REQUEST_CODE";
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        List<String> permissions = Collections.emptyList();
+        if (getIntent().hasExtra(FACEBOOK_PERMISSIONS_EXTRA)) {
+            permissions = getIntent().getStringArrayListExtra(FACEBOOK_PERMISSIONS_EXTRA);
+        }
+        LoginManager.getInstance().logInWithReadPermissions(this, permissions);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (FacebookSdk.isFacebookRequestCode(requestCode)) {
+            data.putExtra(PROVIDER_REQUEST_CODE_EXTRA, requestCode);
```

Comment:
This lib is to be used with Auth0.Lock. When Lock needs to authenticate with a `facebook` connection, it will use this provider and call it passing a custom request code. The problem is that since a few versions, FacebookSDK no longer allows us to customize the request code on the SDK initialization. Previous to this, another issue was that if the SDK was already initialized you couldn't be able to set a custom `request_code` either.


This PR proposes using a simple activity (with no layout) to pass Lock's `request_code` to the `FacebookSignInDelegateActivity`, which will call Facebook login with whatever `request_code` the SDK decides to set. Then when a result is obtained, it will redirect it back using Lock's `request_code`. So it fixes both issues:
- Setting a custom request_code (set by Lock)
- Checking if the sdk was already initialized. (FYI. Now the SDK "doesn't need to be initialized")